eZ/Publish/Core/MVC/Symfony/Routing/Generator/UrlAliasGenerator.php

```diff
@@ -84,15 +84,18 @@ public function __construct(Repository $repository, RouterInterface $defaultRout
     public function doGenerate($location, array $parameters)
     {
         $urlAliasService = $this->repository->getURLAliasService();
-        if (isset($parameters['siteaccess'])) {
+
+        $siteaccess = isset($parameters['siteaccess']) ? $parameters['siteaccess'] : null;
+        if ($siteaccess) {
             // We generate for a different SiteAccess, so potentially in a different language.
             $languages = $this->configResolver->getParameter('languages', null, $parameters['siteaccess']);
             $urlAliases = $urlAliasService->listLocationAliases($location, false, null, null, $languages);
             // Use the target SiteAccess root location
-            $rootLocationId = $this->configResolver->getParameter('content.tree_root.location_id', null, $parameters['siteaccess']);
+            $rootLocationId = $this->configResolver->getParameter('content.tree_root.location_id', null, $siteaccess);
             unset($parameters['siteaccess']);
```

Comment:
Should move this before the if block now that this block is only executed when the value is set, and has a value